ado-agents.tf

```diff
@@ -0,0 +1,215 @@
+          }
+
+          env {
+            name  = "AZP_POOL"
+            value = var.ado-agents_config.poolname
+          }
+
+          env {
+            name = "AZP_TOKEN"
+
+            value_from {
+              secret_key_ref {
+                name = var.ado-agents_config.secretname
+                key  = var.ado-agents_config.secretkey
+              }
+            }
+          }
+
+          env {
+            name  = "namespace"
+            value = var.ado-agents_config.namespace
+          }
+
+          env {
+            name  = "java_version"
+            value = each.value.java_version
+          }
+
+          env {
+            name  = "identifier"
+            value = each.value.identifier
+          }
+
+          termination_message_path   = "/dev/termination-log"
+          termination_message_policy = "File"
+          image_pull_policy          = "Always"
+        }
+
+        restart_policy                   = "Never"
+        termination_grace_period_seconds = 30
+        dns_policy                       = "ClusterFirst"
+      }
+    }
+
+  }
+  wait_for_completion = false
+
+  depends_on = [
+    time_sleep.wait_for_aks_api_dns_propagation,
+    kubernetes_namespace.ado-agents_namespace,
+    kubernetes_secret.azdevops,
+    helm_release.keda_install
+  ]
+}
+
+#
+# var.ado-agents_config.enable true AND var.ado-agents_config.scaledjob set to True so Install ScaledJob not Job
+# https://github.com/kedacore/keda-docs/blob/main/content/docs/2.14/scalers/azure-pipelines.md
+#  https://github.com/kedacore/keda-docs/blob/main/content/docs/2.14/scalers/azure-pipelines.md#example-for-scaledobject
+
+resource "kubectl_manifest" "azdevops_scaledjob_triggerauth" {
+  count     = (var.ado-agents_config.enable && var.ado-agents_config.scaledjob) ? 1 : 0
+  yaml_body = <<YAML
+---
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: pipeline-trigger-auth
+  namespace: ${var.ado-agents_config.namespace}
+spec:
+  secretTargetRef:
+    - parameter: personalAccessToken
+      name: "${var.ado-agents_config.secretname}"
+      key: "${var.ado-agents_config.secretkey}"
+YAML
+  depends_on = [
+    time_sleep.wait_for_aks_api_dns_propagation,
+    kubernetes_namespace.ado-agents_namespace,
+    kubernetes_secret.azdevops,
+    helm_release.keda_install
+  ]
+  force_new = true
+}
+
+resource "kubectl_manifest" "azdevops_agent" {
+  for_each  = (var.ado-agents_config.enable && var.ado-agents_config.scaledjob) ? { for agent in var.ado-agents_config.agents : agent.agent_name => agent } : {}
+  yaml_body = <<YAML
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledJob
+metadata:
+  name: ${var.environment}-${var.aks_cluster_set}-${var.aks_cluster_number}-${each.value.agent_name}
+  namespace: ${var.ado-agents_config.namespace}
+spec:
+  triggers:
+    - type: azure-pipelines
+      metadata:
+        poolName: "${var.ado-agents_config.poolname}"
+        organizationURLFromEnv: "AZP_URL"
```

Comment:
it would be better to create manifest file for scalejob.. refer the ticket.. also you will have to use demands here as well..